guides/customization/node.md

```diff
@@ -0,0 +1,236 @@
+SSH path here since Coder workspaces can clone on behalf of SSh.
+
+This is done in our configure script:
+
+```sh
+# configure
+
+# 1. ensure we have GitHub's host key added to known_hosts
+if ! ssh-keygen -F github.com > /dev/null; then
+  ssh-keyscan github.com >> ~/.ssh/known_hosts 2> /dev/null
+fi
+
+# 2. clone our project
+GIT_REMOTE="git@github.com:bpmct/company-blog.git"
+CLONE_TO="$HOME/company-blog"
+
+if [ ! -d $CLONE_TO ]; then
+  echo "$CLONE_TO is empty... Cloning"
+  git clone $GIT_REMOTE $CLONE_TO/.
+else
+  echo "$CLONE_TO exists ✅"
+fi
+```
+
+If you have multiple repositories, you can replicate part two multiple times in
+the script to clone each repository.
+
+## 4. Add environment variables
+
+There are two ways to add environment variables for everyone using the image.
+
+Option 1: Specify environment variables for your projects in the Dockerfile.
+
+> Note: If you have secret credentials, make sure you make the image private in
+> the Docker Hub, or use a private registry!
+
+```Dockerfile
+# Dockerfile
+
+...
+
+ENV PORT=3000
+
+ENV API_KEY=asdfjkl
+```
+
+Option 2: Add a secrets manager to the Coder Image
+
+You can install an industry-standard secrets manager inside of Coder, just as
+you would on a local machine. This approach is more secure, as it doesn’t
+require you to commit sensitive information, but you’ll need to create an
+account or host one of these services yourself.
+
+For instructions on adding these tools to your Dockerfiles, see these guides:
+
+- [Doppler](https://docs.doppler.com/docs/enclave-installation-docker#option-1-dockerfile)
+- [SecretHub.io](https://secrethub.io/docs/reference/cli/install/#linux)
+- [Hashicorp Vault](https://learn.hashicorp.com/tutorials/vault/getting-started-install#install-vault)
+
+## 5. Build and push your image
+
+You can now build and push your image to use in Coder! We’ll be using Docker
+Hub, but you can use any registry.
+
+```sh
+# log in to Docker Hub
+docker login
+
+# build the image (from the current directory) and tag it
+docker build . -t [username]/gatsby-blog
+
+# push the image
+# note: if you want this image to be private,
+# make sure you create the private image in hub.docker.com
+# or use another private registry
+docker push [username]/gatsby-blog
+```
+
+We recommend you add CI to your git repository so that you can build images
+automatically and source control your image. This can be done with GitHub
+Actions, GitLab, CircleCi, or even Docker Hub itself.
+
+## 6. Add the image to Coder and launch workspace
+
+You can now launch your workspace containing all of the required tools and
+dependencies.
+
+Check out [the example repository](https://github.com/bpmct/company-blog) to see
+an example project, and the image configured for Coder.
+
+---
+
+This guide is maintained by [Ben Potter](https://twitter.com/bpmct). If you have
+questions, feel free to
+[ask on our Slack channel](https://cdr.co/join-community).
```

Comment:
this can absolutely be removed. I was thinking of ways to separate ownership of guides and make it easier for users to give feedback